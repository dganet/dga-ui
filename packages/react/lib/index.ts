import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}