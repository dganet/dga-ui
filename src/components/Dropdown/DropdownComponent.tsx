import React, { useState } from 'react'

import { useTheme } from '../../hooks/useTheme'
import {
    Container,
    DropdownContainerItem,
    DropdownItem,
    DropdownLabel,
    DropdownTitle
} from './DropdownComponent.styles'
import {
    DropdownProps,
    DropdownItem as DropdownItemType
} from './DropdownComponent.types'
import { IoIosArrowDown, IoIosArrowUp } from 'react-icons/io'

const DropdownComponent = ({
    items = [],
    title,
    multSelect = false,
    selectedItem,
    label,
    setSelectedItem
}: DropdownProps) => {
    const [open, setOpen] = useState(false)

    const theme = useTheme()

    const handleClick = () => {
        setOpen(!open)
    }
    const handleItemClick = (item: DropdownItemType) => {
        setSelectedItem(item)
        setOpen(false)
    }

    return (
        <>
            <DropdownLabel theme={theme}>{label}</DropdownLabel>
            <Container theme={theme} onClick={handleClick}>
                <DropdownTitle theme={theme}>
                    {selectedItem?.value ? selectedItem.value : title}
                </DropdownTitle>
                {open ? (
                    <IoIosArrowUp title="arrow-up" />
                ) : (
                    <IoIosArrowDown title="arrow-down" />
                )}

                <DropdownContainerItem theme={theme} isOpen={open}>
                    {items.map(item => (
                        <DropdownItem
                            theme={theme}
                            key={item.id}
                            onClick={() => handleItemClick(item)}
                        >
                            {item.value}
                        </DropdownItem>
                    ))}
                </DropdownContainerItem>
            </Container>
        </>
    )
}
export default DropdownComponent
