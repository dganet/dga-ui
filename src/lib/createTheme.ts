import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
