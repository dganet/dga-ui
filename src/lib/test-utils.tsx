import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
