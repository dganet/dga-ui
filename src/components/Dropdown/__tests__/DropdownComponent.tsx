import { IoIosArrowDown, IoIosArrowUp } from 'react-icons/io'

import { fireEvent, render } from '../../../lib/test-utils'
import DropdownComponent from '../DropdownComponent'
import { DropdownItem } from '../DropdownComponent.types'

describe('<DropdownComponent />', () => {
    it('should render the component', () => {
        const { container } = render(
            <DropdownComponent
                selectedItem={{} as DropdownItem}
                setSelectedItem={() => {}}
                title="Componente Teste"
                items={[
                    {
                        id: 1,
                        value: 'Testando 123...'
                    }
                ]}
            />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should be able to click on component', () => {
        const itemValue = 'Testando 123...'

        const { container, getByTitle, getByText } = render(
            <DropdownComponent
                selectedItem={{} as DropdownItem}
                setSelectedItem={() => {}}
                title="Componente Teste"
                items={[
                    {
                        id: 1,
                        value: itemValue
                    }
                ]}
            />
        )

        const arrowDown = getByTitle('arrow-down')
        const dropdownOption = getByText(itemValue)

        expect(arrowDown).toBeInTheDocument()

        fireEvent.click(dropdownOption)

        expect(arrowDown).not.toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })
})
