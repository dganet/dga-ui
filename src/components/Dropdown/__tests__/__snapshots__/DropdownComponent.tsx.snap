// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownComponent /> should be able to click on component 1`] = `
<span
  class="sc-eCstlR dhneKu"
/>
`;

exports[`<DropdownComponent /> should render the component 1`] = `
<span
  class="sc-eCstlR dhneKu"
/>
`;
